proxy/InkAPI.cc

```diff
@@ -8256,6 +8265,17 @@ TSHttpTxnConfigStringSet(TSHttpTxn txnp, TSOverridableConfigKey conf, const char
     if (value && length > 0) {
       s->t_state.txn_conf->client_cert_filepath = const_cast<char *>(value);
     }
+  case TS_CONFIG_HTTP_RESPONSE_VIA_TRANSPORT:
+    data._s.setView(value, length);
+    if (ViaTransportVerbosityConversion(s->t_state.txn_conf->response_via_transport, "proxy.config.http.response_via_tranport",
+                                        RECD_STRING, data)) {
+    }
+    break;
+  case TS_CONFIG_HTTP_REQUEST_VIA_TRANSPORT:
+    data._s.setView(value, length);
+    if (ViaTransportVerbosityConversion(s->t_state.txn_conf->request_via_transport, "proxy.config.http.request_via_tranport",
```

Comment:
It feels like we ought to have a way to avoid up to 3x strncasecmp() for every time I need to set a configuration to a specific value. Maybe we'd want to have a TS API, that lets you do the lookup once (on config load), such that you go from "full" -> TS_VIA_TRANSPORT_FULL once and for all, and from there on, just allow it to set it to TS_VIA_TRANSPORT_FULL (2) without the 3x strncasecmp()'s.